client/client_webhook.py

```diff
@@ -83,69 +82,29 @@ def process_webhook(self):
-            if not os.path.isfile(manifest_path):
-                manifest_path = None
-        meta_path = os.path.join(repo_dir, 'meta.json')
-        meta = None
-        if os.path.isfile(meta_path):
-            meta = MetaData(meta_path)
-        manifest = Manifest(file_name=manifest_path, repo_name=repo_name, files_path=repo_dir, meta=meta)
-
-        # determining how the repo was generated/created:
-        generator = ''
-        if manifest.generator and manifest.generator['name'] and manifest.generator['name'].startswith('ts'):
-            generator = 'ts'
-        if not generator:
-            dirs = sorted(get_subdirs(repo_dir, True))
-            if 'content' in dirs:
-                repo_dir = os.path.join(repo_dir, 'content')
-            elif 'usfm' in dirs:
-                repo_dir = os.path.join(repo_dir, 'usfm')
-
-        manifest_path = os.path.join(repo_dir, 'manifest.json')
-        write_file(manifest_path, manifest.__dict__)  # Write it back out so it's using the latest manifest format
-
-        input_format = manifest.format
-        resource_type = manifest.resource['id']
-        if resource_type == 'ulb' or resource_type == 'udb':
-            resource_type = 'bible'
+        # Get the resource container
+        rc = RC(repo_dir, repo_name)
 
-        try:
-            preprocessor_class = self.str_to_class(
-                'preprocessors.{0}{1}{2}Preprocessor'.format(generator.capitalize(),
-                                                             resource_type.capitalize(),
-                                                             input_format.capitalize()))
-        except AttributeError as e:
-            self.logger.info('Got AttributeError: {0}'.format(e.message))
-            preprocessor_class = preprocessors.Preprocessor
-
-        # merge the source files with the template
+        # Preprocess the files
         output_dir = tempfile.mkdtemp(dir=self.base_temp_dir, prefix='output_')
-        preprocessor = preprocessor_class(manifest, repo_dir, output_dir)
-        preprocessor.run()
+        do_preprocess(rc, repo_dir, output_dir)
 
         # 3) Zip up the massaged files
         # context.aws_request_id is a unique ID for this lambda call, so using it to not conflict with other requests
         zip_filename = commit_id + '.zip'
         zip_filepath = os.path.join(self.base_temp_dir, zip_filename)
         self.logger.info('Zipping files from {0} to {1}...'.format(output_dir, zip_filepath))
         add_contents_to_zip(zip_filepath, output_dir)
-        if os.path.isfile(manifest_path) and not os.path.isfile(os.path.join(output_dir, 'manifest.json')):
-            add_file_to_zip(zip_filepath, manifest_path, 'manifest.json')
         self.logger.info('finished.')
 
         # 4) Upload zipped file to the S3 bucket
-        s3_handler = self.s3_handler_class(self.pre_convert_bucket)
+        s3_handler = S3Handler(self.pre_convert_bucket)
```

Comment:
You can mock a function of an existing class, change its return value, etc, as you probably know.